### ISSUE TITLE
Hyperparameters optimization to get a better performance at overlap part

### ISSUE THREAD

**Issue by Tingyan-Guo** [earthspecies/voxaboxen#25]
Hi there,

Thank you for developing this wonderful framework! I've been working with a dataset containing highly overlapping vocalizations where I'm seeing good detection performance on non-overlapping segments but frequent misses on overlapping portions.

Would you have any recommendations for parameter adjustments that might specifically improve performance on overlapping segments? I tried using mixup augmentation via the training command line, but it doesn't seem to perform better.

Great appreciation for any suggestions you may have : )

The following example figures are from one training with:
```
bidirectional: true
comb_discard_thresh: 0.75
detection_threshold: 0.2
early_stopping: false
encoder_type: beats
mixup: false
model_selection_class_threshold: 0.0
n_epochs: 50
n_map: 0
n_val_fit: 19
peak_distance: 5
sr: 16000
```
The following two pictures are spectrograms with inference annotations, which seem to miss some annotations in the overlapping call sections where two animals vocalized simultaneously.

<img width="1103" height="704" alt="Image" src="https://github.com/user-attachments/assets/d03ccea5-9596-4dad-a6ff-1af2c39c491a" />

<img width="1092" height="713" alt="Image" src="https://github.com/user-attachments/assets/3cbb6e61-7ca7-4953-8b48-6f84c47d7191" />
And the train history and change track of learning rate:
<img width="971" height="668" alt="Image" src="https://github.com/user-attachments/assets/c6afa318-1699-4fad-b251-5b8ed70c4b03" />

**Comment by Tingyan-Guo** [comment]
The test set performance results are as follows:
macro-f1@0.5: 0.8291430739159613
macro-f1@0.8: 0.5849394707580031
micro-f1@0.5: 0.8254091971940763
micro-f1@0.8: 0.6196414653156664
seg-macro-f1@0.5: 0.8056824465484779
seg-macro-f1@0.8: 0.8056824465484779
seg-micro-f1@0.5: 0.8179623983739837
seg-micro-f1@0.8: 0.8179623983739837
I hope this additional information helps

**Comment by benjaminsshoffman** [comment]
Thanks for reaching out. Here are some suggestions:

- I noticed a bug if using `mixup` together with `bidirectional`, which should now be fixed
- Since your events are short, it could reduce overfitting to train with a smaller window size (`--clip-duration=3`), for instance
- Since your events are highly overlapped, you could try turning off non-maximal suppression (`--nms=none`)
- Similarly, you could reduce the time that might elapse between event starts, by setting `peak-distance=3` or lower (min is 1).
- For short events, sometimes it also helps to feed the audio in at half-speed, which is accomplished by setting `--sr=32000`
- For training, I would recommend sweeping a few learning rates and choosing the best one on a validation set.

Finally, it could be worth experimenting with the segmentation-based approach (`--segmentation-based --rho=1`). Note this is incompatible with the bidirectional setting.

Hopefully this improves things, please let me know if you find anything that helps out.

**Comment by benjaminsshoffman** [comment]
Also, not sure if you're doing this but I would train for longer than 8 epochs (I'd suggest 20+)
